File: 2015/03/julia/03.jl
input = split(readlines(stdin)[1], "")

data = vcat([[0, 0]], map(input) do char
    char == ">" && return [1, 0]
    char == "v" && return [0, -1]
    char == "<" && return [-1, 0]
    char == "^" && return [0, 1]
end)

println(length(unique(cumsum(data))))
println(length(unique([cumsum(data[1:2:end]); cumsum(data[2:2:end])])))

File: 2015/03/R/03.R
input <- as.character(strsplit(scan("stdin", "character", quiet = TRUE), "")[[1L]])

dirs <- list(
  "^" = c(0, 1),
  "v" = c(0, -1),
  "<" = c(-1, 0),
  ">" = c(1, 0)
)

coords <- rbind(c(0, 0), do.call(rbind, dirs[input]))

cat(nrow(unique(apply(coords, 2L, cumsum))), "\n")

cat(nrow(unique(rbind(
  apply(coords[c(TRUE, FALSE),], 2L, cumsum), 
  apply(coords[c(FALSE, TRUE),], 2L, cumsum)))), "\n")

